geode-core/src/main/java/org/apache/geode/internal/cache/TXRegionState.java

```diff
@@ -319,14 +319,24 @@ void buildMessage(InternalRegion r, TXCommitMessage msg) {
     try {
       if (!r.getScope().isLocal() && !this.entryMods.isEmpty()) {
 
-        msg.startRegion(r, entryMods.size());
+        TXCommitMessage.RegionCommit regionCommit = msg.startRegion(r, entryMods.size());
         Iterator it = this.entryMods.entrySet().iterator();
-        Set<InternalDistributedMember> newMemberSet = new HashSet<InternalDistributedMember>();
+        Set<InternalDistributedMember> newMemberSet = new HashSet();
+        Set<InternalDistributedMember> redundantMemberSet =
+            new HashSet();
 
         if (r.getScope().isDistributed()) {
           DistributedRegion dr = (DistributedRegion) r;
           msg.addViewVersion(dr, dr.getDistributionAdvisor().startOperation());
           newMemberSet.addAll(dr.getCacheDistributionAdvisor().adviseTX());
+          redundantMemberSet.addAll(dr.getCacheDistributionAdvisor().adviseTX());
```

Comment:
Rather than two calls to `dr.getCacheDistributionAdvisor().adviseTX()`, could this be extracted to a local variable?